### PR TITLE
Make the nuclear beer keg react to the NAD.

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
Using the NAD on the beer keg nuke will cause it to explode in a shower of beer.

## Why It's Good For The Game
It's funny.

## Testing
Used other objects, attacked normally.
Used NAD, got drunk.

## Changelog
:cl:
add: Some hooligan added a self destruct device to that beer keg we dressed up like the nuke. Fortunately, it can only be set off by the real nuclear authentication disk, and nobody would do that, right?
/:cl:

### CODE REVIEW

File: code/modules/reagents/reagent_dispenser.dm [hunk 1]
```diff
@@ -279,6 +279,29 @@
 	icon = 'icons/obj/nuclearbomb.dmi'
 	icon_state = "nuclearbomb0"
 	anchored = TRUE
+	var/exploding = FALSE
+
+/obj/structure/reagent_dispensers/beerkeg/nuke/attackby(obj/item/O as obj, mob/user as mob, params)
```
Comment (Henri215) [inline code/modules/reagents/reagent_dispenser.dm:284]:
```suggestion
/obj/structure/reagent_dispensers/beerkeg/nuke/attackby(obj/item/O, mob/user, params)
```

File: code/modules/reagents/reagent_dispenser.dm [hunk 2]
```diff
@@ -279,6 +279,29 @@
...
+		"<span class='danger'>You prime [src] with [O] and it begins making a loud beeping noise! Better run!</span>")
+	playsound(src, 'sound/machines/alarm.ogg', 100, 0, 0)
+	exploding = TRUE
+	sleep(13 SECONDS)
+	var/datum/reagents/R = new(100)
+	R.my_atom = src
+	R.add_reagent("beer", 100)
+	var/datum/effect_system/smoke_spread/chem/smoke = new
+	smoke.set_up(R, src, TRUE)
+	playsound(src.loc, 'sound/effects/smoke.ogg', 50, 1, -3)
```
Comment (Henri215) [inline code/modules/reagents/reagent_dispenser.dm:301]:
```suggestion
	playsound(src, 'sound/machines/alarm.ogg', 100, FALSE, 0)
	exploding = TRUE
	sleep(13 SECONDS)
	var/datum/reagents/R = new(100)
	R.my_atom = src
	R.add_reagent("beer", 100)
	var/datum/effect_system/smoke_spread/chem/smoke = new
	smoke.set_up(R, src, TRUE)
	playsound(src.loc, 'sound/effects/smoke.ogg', 50, TRUE, -3)
```

File: code/modules/reagents/reagent_dispenser.dm [hunk 3]
```diff
@@ -279,6 +279,29 @@
...
+	if(exploding)
+		return
+	if(!istype(O, /obj/item/disk/nuclear))
+		return
+	user.visible_message(
+		"<span class='danger'>[user] inserts [O] into [src] and it begins making a loud beeping noise! Uh-oh!</span>",
+		"<span class='danger'>You prime [src] with [O] and it begins making a loud beeping noise! Better run!</span>")
+	playsound(src, 'sound/machines/alarm.ogg', 100, 0, 0)
+	exploding = TRUE
+	sleep(13 SECONDS)
```
Comment (Henri215) [inline code/modules/reagents/reagent_dispenser.dm:295]: Can be done with a ```addtimer()``` here

File: code/modules/reagents/reagent_dispenser.dm [hunk 4]
```diff
@@ -279,6 +279,31 @@
 	icon = 'icons/obj/nuclearbomb.dmi'
 	icon_state = "nuclearbomb0"
 	anchored = TRUE
+	var/exploding = FALSE
```
Comment (Henri215) [inline code/modules/reagents/reagent_dispenser.dm:282]:
```suggestion
	/// If TRUE, prevents the player from inserting the disk again while it is currently exploding.
	var/exploding = FALSE
```

**Review by Henri215 — APPROVED** [review]
LGTM

File: code/modules/reagents/reagent_dispenser.dm [hunk 5]
```diff
@@ -279,6 +279,32 @@
...
+/obj/structure/reagent_dispensers/beerkeg/nuke/proc/explode()
+	var/datum/reagents/R = new(100)
+	R.my_atom = src
+	R.add_reagent("beer", 100)
+	var/datum/effect_system/smoke_spread/chem/smoke = new
+	smoke.set_up(R, src, TRUE)
+	playsound(src.loc, 'sound/effects/smoke.ogg', 50, TRUE, -3)
+	smoke.start(3)
+	qdel(R)
+	qdel(src)
```
Comment (DGamerL) [inline code/modules/reagents/reagent_dispenser.dm:307]: Maybe make it do a small explosion aswell?

Comment (FunnyMan3595) [reply code/modules/reagents/reagent_dispenser.dm:307]: Nah, I want it to just be a funny thing that gets people drunk, not something that can cause actual damage/harm, even small.

Comment (DGamerL) [reply code/modules/reagents/reagent_dispenser.dm:307]: Fair, then it might also be good to not qdel it? So you can still get more drunk with it

**Review by Contrabang — APPROVED** [review]
Code is fine, I'd like more votes on this before merging

**Review by DGamerL — APPROVED** [review]

**Pull request merged**